### PR TITLE
cmake: Remove implict ccache/compile_commands.json

### PR DESCRIPTION
**Pull request by juan-lunarg**
Remove CMake code for
- USE_CCACHE
- CMAKE_EXPORT_COMPILE_COMMANDS

Show in the docs how to enable these features via cli or environment variables.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -110,14 +108,6 @@ add_library(Vulkan-Headers INTERFACE)
 target_include_directories(Vulkan-Headers INTERFACE ${VulkanHeaders_INCLUDE_DIRS})
 add_library(Vulkan::Headers ALIAS Vulkan-Headers)
 
-option(USE_CCACHE "Use ccache" OFF)
-if(USE_CCACHE)
```
Comment (spencer-lunarg) [inline CMakeLists.txt:114]: worth leaving a 

```
if (USE_CCACHE)
   don't use anymore warning
endif()
```

for a while (I know I might need a reminder) or will cmake let me know that `-DUSE_CCACHE` didn't work

Comment (ncesario-lunarg) [reply CMakeLists.txt:114]: CMake _should_ let you know that a variable you passed in was not used (at least it used to).

Comment (spencer-lunarg) [reply CMakeLists.txt:114]: perfect!

**Review by ncesario-lunarg — APPROVED** [review]

File: BUILD.md [hunk 2]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
 | BUILD_WSI_XCB_SUPPORT | Linux | `ON` | Build the components with XCB support. |
 | BUILD_WSI_XLIB_SUPPORT | Linux | `ON` | Build the components with Xlib support. |
 | BUILD_WSI_WAYLAND_SUPPORT | Linux | `ON` | Build the components with Wayland support. |
-| USE_CCACHE | Linux | `OFF` | Enable caching with the CCache program. |
+
+### CCACHE
+
+There are 2 methods to enable CCACHE
```
Comment (lunarpapillo) [inline BUILD.md:308]: Append a colon.

File: BUILD.md [hunk 3]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+
+1.) Set environment variables
+
+```
+# Requires CMake 3.17 (https://cmake.org/cmake/help/latest/envvar/CMAKE_LANG_COMPILER_LAUNCHER.html)
+export CMAKE_CXX_COMPILER_LAUNCHER="/usr/bin/ccache"
+export CMAKE_C_COMPILER_LAUNCHER="/usr/bin/ccache"
```
Comment (lunarpapillo) [inline BUILD.md:315]: The double-quotes aren't necessary.

File: BUILD.md [hunk 4]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+```
+
+2.) Pass in cache variables
+
+```
+cmake ... -D CMAKE_CXX_COMPILER_LAUNCHER=/usr/bin/ccache CMAKE_C_COMPILER_LAUNCHER=/usr/bin/ccache
```
Comment (lunarpapillo) [inline BUILD.md:321]: The second variable also needs a `-D`, right?

File: BUILD.md [hunk 5]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+```
+
+### EXPORT_COMPILE_COMMANDS
+
+There are 2 methods to enable exporting compile commands
```
Comment (lunarpapillo) [inline BUILD.md:326]: Append a colon.

File: BUILD.md [hunk 6]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+
+1.) Set environment variables
+
+```
+# Requires CMake 3.17 (https://cmake.org/cmake/help/latest/envvar/CMAKE_EXPORT_COMPILE_COMMANDS.html)
+export CMAKE_EXPORT_COMPILE_COMMANDS="ON"
```
Comment (lunarpapillo) [inline BUILD.md:332]: Double quotes are not necessary.

File: BUILD.md [hunk 7]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+```
+
+2.) Pass in cache variables
+
+```
+cmake ... -D CMAKE_EXPORT_COMPILE_COMMANDS=ON
+```
+
+NOTE: Modern tools will generally enable exporting compile commands for you (EX: VSCode).
+Also CMAKE_EXPORT_COMPILE_COMMANDS is implemented only by Makefile and Ninja generators. For other generators, this option is ignored.
```
Comment (lunarpapillo) [inline BUILD.md:342]: `CMAKE_EXPORT_COMPILE_COMMANDS` in the prose should be in backquotes.

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
I was asked to provide a CI opinion, so here I am reviewing. :-)

CI doesn't use `CCACHE`, so this should not affect us.

It may affect Linux developers, but if the curator is okay with it, so am I.

File: BUILD.md [hunk 8]
```diff
@@ -302,7 +302,44 @@ on/off options currently supported by this repository:
...
+export CMAKE_EXPORT_COMPILE_COMMANDS="ON"
+```
+
+2.) Pass in cache variables
+
+```
+cmake ... -D CMAKE_EXPORT_COMPILE_COMMANDS=ON
+```
+
+NOTE: Modern tools will generally enable exporting compile commands for you (EX: VSCode).
```
Comment (lunarpapillo) [inline BUILD.md:341]: "e.g." is probably better than "EX:".

**Pull request merged**